src/components/utils/onfidoApi.ts

```diff
@@ -218,6 +222,79 @@ export const requestChallenges = (
   performHttpReq(options, onSuccess, (request) => formatError(request, onError))
 }
 
+/* v4 APIs */
+export const uploadBinaryMedia = (
+  { file, filename, sdkMetadata }: UploadDocumentPayload,
+  url: string | undefined,
+  token: string | undefined,
+  includeHmacAuth = false
+): Promise<UploadBinaryMediaReponse> =>
+  new Promise((resolve, reject) => {
+    try {
+      const tokenData = parseJwt(token)
+      const formData = new FormData()
+      formData.append('media', file, filename)
```

Comment:
Do we make sure somewhere the filename always exists or has a default?

Reply:
Not yet. Does `filename` required for the APIs? :thinking:

Reply:
It's not required but clients have been complaining of docs with no filenames lately. I introduced [a default name](https://github.com/onfido/onfido-sdk-ui/blob/master/src/components/Confirm/Confirm.js#L229) for standard documents a few weeks ago.